ingest/sqs/service/node_status_checker.go

```diff
@@ -0,0 +1,39 @@
+package service
+
+import (
+	"github.com/cosmos/cosmos-sdk/client"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v24/ingest/sqs/domain"
+)
+
+type nodeStatusChecker struct {
+	// example format: tcp://localhost:26657
+	address string
+}
+
+var _ domain.NodeStatusChecker = (*nodeStatusChecker)(nil)
+
+func NewNodeStatusChecker(address string) domain.NodeStatusChecker {
+	return &nodeStatusChecker{
+		address: address,
+	}
+}
+
+// IsNodeSyncing implements NodeStatusChecker.
+func (n *nodeStatusChecker) IsNodeSyncing(ctx sdk.Context) (bool, error) {
+	// Get client
+	client, err := client.NewClientFromNode(n.address)
+	if err != nil {
+		return false, err
+	}
+
+	// Get status
+	statusResult, err := client.Status(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	// Return if node is catching up
+	return statusResult.SyncInfo.CatchingUp, nil
```

Comment:
Maybe in future iterations this could also check if we are connected to at least one peer, just for some slightly higher resiliency. Also could maybe store the last height we checked and the timestamp on that, and if the next time we call the height is the same with a delta of x seconds, this also triggers as false.

Reply:
Is there any trivial way you know of for checking the number of peers connected?

> Also could maybe store the last height we checked and the timestamp on that, and if the next time we call the height is the same with a delta of x seconds, this also triggers as false. 

We have this check in the web server healthcheck.

Given that here, this mechanism is only used at cold start, I don't think that we need the same level of resiliency.

I've previously had this check at this level of abstraction as well. However, it has proven to add meaningful complexity without direct benefit. That's the only reason I'm proposing against it today